Prepare for v1.0.0 release.

jonathanknowles opened the pull request:
## Related Issue

#19 

## Summary:

This PR:

- [x] Removes redundant dependencies from `package.yaml`.
- [x] Updates the change log with the correct release date.
- [x] The "Added", "Changed", and "Removed" headings do not apply to the initial release, and so have been removed from the change log. (These sections will be added in subsequent releases.)
- [x] Updates `stack.yaml` to the most recent LTS.

jonathanknowles commented:
@rvl wrote:
> It will probably cause less trouble if we keep libs that will be used by cardano-wallet on the same Stackage LTS series as cardano-wallet.

That's a good point.

As `cardano-wallet` currently uses `lts-14.25`, we could use this version. However, any change we make is likely to become out of date very quickly.

We could change this entry to link to https://raw.githubusercontent.com/input-output-hk/cardano-haskell/master/snapshots/cardano-1.11.0.yaml. Can you see any disadvantages to this approach?

rvl commented:
Main problem with LTS 15.x is that it's using GHC 8.8 when cardano-wallet is on GHC 8.6.

I think it's enough to keep on the same LTS major version (14.x) as cardano-wallet. For this package, the cardano-haskell snapshot has too many irrelevant packages.

jonathanknowles commented:
@rvl wrote:
> Main problem with LTS 15.x is that it's using GHC 8.8 when cardano-wallet is on GHC 8.6.
> 
> I think it's enough to keep on the same LTS major version (14.x) as cardano-wallet. For this package, the cardano-haskell snapshot has too many irrelevant packages.

Okay! I've changed it to use `lts-14.25`. (I did experiment with using the same resolver reference as `cardano-wallet`, but that caused stack to start pulling in dozens of extra unnecessary packages, as predicted.)

Fixed in d4374a225ad77366735998e8115216273172d335.
